kuksa-sdk/src/main/kotlin/org/eclipse/kuksa/DataBrokerConnection.kt

```diff
@@ -45,6 +45,17 @@ class DataBrokerConnection internal constructor(
     private val managedChannel: ManagedChannel,
     private val defaultDispatcher: CoroutineDispatcher = Dispatchers.Default,
 ) {
+    val disconnectListeners = MultiListener<DisconnectListener>()
+
+    init {
+        val state = managedChannel.getState(false)
+        managedChannel.notifyWhenStateChanged(state) {
+            val listeners = disconnectListeners.get()
```

Comment:
using "by" / ReadOnlyProperty will lead to the following issues: 
a) I can only access the list anymore (no more register / unregister methods) 
b) get() method is no longer internal but public and we can no longer apply @JvmSynthetic annotation
c) MultiListener would have a hard dependency on DataBrokerConnection to make it work:
`class MultiListener<T> : ReadOnlyProperty<DataBrokerConnection, List<T>> {` 

This might be mostly a naming conflict or as mentioned above solved by removing the unusualness by simply exposing corresponding methods from it. Depending on your decision if we keep MultiListener I will invest time here or not